// baseui/map-marker/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.anchorSize = exports.dragShadowWidth = exports.dragShadowMarginTop = exports.dragShadowHeight = exports.FLOATING_MARKER_ANCHOR_TYPES = exports.FLOATING_MARKER_ANCHOR_POSITIONS = exports.FLOATING_MARKER_SIZES = exports.PINHEAD_TYPES = exports.PINHEAD_DIMENSIONS = exports.xSmallPinheadDimension = exports.PINHEAD_SIZES_SHAPES = exports.NEEDLE_HEIGHTS = exports.NEEDLE_SIZES = void 0;

var _Object$freeze, _Object$freeze2;

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

/*
Copyright (c) Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
var NEEDLE_SIZES = Object.freeze({
  none: 'none',
  short: 'short',
  medium: 'medium',
  tall: 'tall'
});
exports.NEEDLE_SIZES = NEEDLE_SIZES;
var NEEDLE_HEIGHTS = Object.freeze((_Object$freeze = {}, _defineProperty(_Object$freeze, NEEDLE_SIZES.none, 0), _defineProperty(_Object$freeze, NEEDLE_SIZES.short, 4), _defineProperty(_Object$freeze, NEEDLE_SIZES.medium, 12), _defineProperty(_Object$freeze, NEEDLE_SIZES.tall, 20), _Object$freeze));
exports.NEEDLE_HEIGHTS = NEEDLE_HEIGHTS;
var PINHEAD_SIZES_SHAPES = Object.freeze({
  xSmallCircle: 'x-small-circle',
  xSmallSquare: 'x-small-square',
  small: 'small',
  medium: 'medium',
  large: 'large'
});
exports.PINHEAD_SIZES_SHAPES = PINHEAD_SIZES_SHAPES;
var xSmallPinheadDimension = {
  height: 16,
  icon: 4
};
exports.xSmallPinheadDimension = xSmallPinheadDimension;
var PINHEAD_DIMENSIONS = Object.freeze((_Object$freeze2 = {}, _defineProperty(_Object$freeze2, PINHEAD_SIZES_SHAPES.xSmallSquare, xSmallPinheadDimension), _defineProperty(_Object$freeze2, PINHEAD_SIZES_SHAPES.xSmallCircle, xSmallPinheadDimension), _defineProperty(_Object$freeze2, PINHEAD_SIZES_SHAPES.small, {
  height: 24,
  icon: 16
}), _defineProperty(_Object$freeze2, PINHEAD_SIZES_SHAPES.medium, {
  height: 36,
  icon: 16
}), _defineProperty(_Object$freeze2, PINHEAD_SIZES_SHAPES.large, {
  height: 48,
  icon: 24
}), _Object$freeze2));
exports.PINHEAD_DIMENSIONS = PINHEAD_DIMENSIONS;
var PINHEAD_TYPES = Object.freeze({
  floating: 'floating',
  fixed: 'fixed'
});
exports.PINHEAD_TYPES = PINHEAD_TYPES;
var FLOATING_MARKER_SIZES = Object.freeze({
  small: 'small',
  medium: 'medium',
  large: 'large'
});
exports.FLOATING_MARKER_SIZES = FLOATING_MARKER_SIZES;
var FLOATING_MARKER_ANCHOR_POSITIONS = Object.freeze({
  none: 'none',
  topLeft: 'top-left',
  topRight: 'top-right',
  bottomRight: 'bottom-right',
  bottomLeft: 'bottom-left'
});
exports.FLOATING_MARKER_ANCHOR_POSITIONS = FLOATING_MARKER_ANCHOR_POSITIONS;
var FLOATING_MARKER_ANCHOR_TYPES = Object.freeze({
  circle: 'circle',
  square: 'square'
});
exports.FLOATING_MARKER_ANCHOR_TYPES = FLOATING_MARKER_ANCHOR_TYPES;
var dragShadowHeight = 4;
exports.dragShadowHeight = dragShadowHeight;
var dragShadowMarginTop = 6;
exports.dragShadowMarginTop = dragShadowMarginTop;
var dragShadowWidth = 6;
exports.dragShadowWidth = dragShadowWidth;
var anchorSize = 16;
exports.anchorSize = anchorSize;
