// baseui/map-marker/fixed-marker.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _index = require("../styles/index.js");

var _overrides = require("../helpers/overrides.js");

var _constants = require("./constants.js");

var _pinHead = _interopRequireDefault(require("./pin-head.js"));

var _styledComponents = require("./styled-components.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { if (typeof Symbol === "undefined" || !(Symbol.iterator in Object(arr))) return; var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

var Needle = function Needle(_ref) {
  var size = _ref.size,
      background = _ref.background,
      _ref$overrides = _ref.overrides,
      overrides = _ref$overrides === void 0 ? {} : _ref$overrides;

  var _getOverrides = (0, _overrides.getOverrides)(overrides.Needle, _styledComponents.Needle),
      _getOverrides2 = _slicedToArray(_getOverrides, 2),
      Needle = _getOverrides2[0],
      needleProps = _getOverrides2[1];

  return /*#__PURE__*/React.createElement(Needle, _extends({
    $background: background,
    $height: _constants.NEEDLE_HEIGHTS[size]
  }, needleProps));
};

var DragShadow = function DragShadow(_ref2) {
  var background = _ref2.background,
      dragging = _ref2.dragging,
      height = _ref2.height,
      _ref2$overrides = _ref2.overrides,
      overrides = _ref2$overrides === void 0 ? {} : _ref2$overrides;

  var _getOverrides3 = (0, _overrides.getOverrides)(overrides.DragShadowContainer, _styledComponents.DragShadowContainer),
      _getOverrides4 = _slicedToArray(_getOverrides3, 2),
      DragShadowContainer = _getOverrides4[0],
      dragShadowContainerProps = _getOverrides4[1];

  var _getOverrides5 = (0, _overrides.getOverrides)(overrides.DragShadow, _styledComponents.DragShadow),
      _getOverrides6 = _slicedToArray(_getOverrides5, 2),
      DragShadow = _getOverrides6[0],
      dragShadowProps = _getOverrides6[1];

  return /*#__PURE__*/React.createElement(DragShadowContainer, _extends({
    $width: _constants.dragShadowWidth,
    $height: height,
    $dragging: dragging
  }, dragShadowContainerProps), /*#__PURE__*/React.createElement(DragShadow, _extends({
    $width: _constants.dragShadowWidth,
    $background: background
  }, dragShadowProps)));
};

var FixedMarker = function FixedMarker(_ref3) {
  var _ref3$size = _ref3.size,
      size = _ref3$size === void 0 ? _constants.PINHEAD_SIZES_SHAPES.medium : _ref3$size,
      _ref3$needle = _ref3.needle,
      needle = _ref3$needle === void 0 ? _constants.NEEDLE_SIZES.medium : _ref3$needle,
      label = _ref3.label,
      startEnhancer = _ref3.startEnhancer,
      endEnhancer = _ref3.endEnhancer,
      color = _ref3.color,
      background = _ref3.background,
      _ref3$dragging = _ref3.dragging,
      dragging = _ref3$dragging === void 0 ? false : _ref3$dragging,
      _ref3$overrides = _ref3.overrides,
      overrides = _ref3$overrides === void 0 ? {} : _ref3$overrides;

  var _useStyletron = (0, _index.useStyletron)(),
      _useStyletron2 = _slicedToArray(_useStyletron, 2),
      theme = _useStyletron2[1];

  var _theme$colors = theme.colors,
      backgroundInversePrimary = _theme$colors.backgroundInversePrimary,
      primaryB = _theme$colors.primaryB;
  color = color || primaryB;
  background = background || backgroundInversePrimary;
  var doesPinHeadTransformOnDrag = needle !== _constants.NEEDLE_SIZES.none;

  var _getOverrides7 = (0, _overrides.getOverrides)(overrides.Root, _styledComponents.FixedMarkerRoot),
      _getOverrides8 = _slicedToArray(_getOverrides7, 2),
      Root = _getOverrides8[0],
      rootProps = _getOverrides8[1];

  var _getOverrides9 = (0, _overrides.getOverrides)(overrides.DragContainer, _styledComponents.FixedMarkerDragContainer),
      _getOverrides10 = _slicedToArray(_getOverrides9, 2),
      FixedMarkerDragContainer = _getOverrides10[0],
      fixedMarkerDragContainerProps = _getOverrides10[1];

  return /*#__PURE__*/React.createElement(Root, _extends({
    "data-baseweb": "fixed-map-marker"
  }, rootProps), /*#__PURE__*/React.createElement(FixedMarkerDragContainer, _extends({
    $translateAmount: _constants.dragShadowMarginTop + _constants.dragShadowHeight,
    $performTranslate: doesPinHeadTransformOnDrag && dragging
  }, fixedMarkerDragContainerProps), /*#__PURE__*/React.createElement(_pinHead.default, _extends({
    size: size,
    label: label
  }, startEnhancer ? {
    startEnhancer: startEnhancer
  } : {}, endEnhancer ? {
    endEnhancer: endEnhancer
  } : {}, {
    color: color,
    background: background,
    type: _constants.PINHEAD_TYPES.fixed,
    overrides: overrides
  })), /*#__PURE__*/React.createElement(Needle, {
    size: needle,
    background: background,
    overrides: overrides
  })), doesPinHeadTransformOnDrag && /*#__PURE__*/React.createElement(DragShadow, {
    background: background,
    dragging: dragging,
    height: _constants.dragShadowMarginTop + _constants.dragShadowHeight,
    overrides: overrides
  }));
};

var _default = FixedMarker;
exports.default = _default;
