// baseui/map-marker/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "FixedMarker", {
  enumerable: true,
  get: function get() {
    return _fixedMarker.default;
  }
});
Object.defineProperty(exports, "FloatingMarker", {
  enumerable: true,
  get: function get() {
    return _floatingMarker.default;
  }
});
Object.defineProperty(exports, "FLOATING_MARKER_ANCHOR_POSITIONS", {
  enumerable: true,
  get: function get() {
    return _constants.FLOATING_MARKER_ANCHOR_POSITIONS;
  }
});
Object.defineProperty(exports, "FLOATING_MARKER_SIZES", {
  enumerable: true,
  get: function get() {
    return _constants.FLOATING_MARKER_SIZES;
  }
});
Object.defineProperty(exports, "FLOATING_MARKER_ANCHOR_TYPES", {
  enumerable: true,
  get: function get() {
    return _constants.FLOATING_MARKER_ANCHOR_TYPES;
  }
});
Object.defineProperty(exports, "NEEDLE_SIZES", {
  enumerable: true,
  get: function get() {
    return _constants.NEEDLE_SIZES;
  }
});
Object.defineProperty(exports, "PINHEAD_SIZES_SHAPES", {
  enumerable: true,
  get: function get() {
    return _constants.PINHEAD_SIZES_SHAPES;
  }
});

var _fixedMarker = _interopRequireDefault(require("./fixed-marker.js"));

var _floatingMarker = _interopRequireDefault(require("./floating-marker.js"));

var _constants = require("./constants.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
