// baseui/map-marker/styled-components.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PinHead = exports.InnerXSmallAnchor = exports.OuterXSmallAnchor = exports.FixedMarkerDragContainer = exports.FixedMarkerRoot = exports.FloatingMarkerAnchorContainer = exports.FloatingMarkerPinHeadContainer = exports.FloatingMarkerRoot = exports.Needle = exports.DragShadow = exports.DragShadowContainer = exports.getAnchorTransform = void 0;

var _index = require("../styles/index.js");

var _constants = require("./constants.js");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var getAnchorTransform = function getAnchorTransform(anchor, anchorSize) {
  var _FLOATING_MARKER_ANCH;

  return (_FLOATING_MARKER_ANCH = {}, _defineProperty(_FLOATING_MARKER_ANCH, _constants.FLOATING_MARKER_ANCHOR_POSITIONS.none, ""), _defineProperty(_FLOATING_MARKER_ANCH, _constants.FLOATING_MARKER_ANCHOR_POSITIONS.topLeft, "translate(".concat(anchorSize, "px, ").concat(anchorSize, "px)")), _defineProperty(_FLOATING_MARKER_ANCH, _constants.FLOATING_MARKER_ANCHOR_POSITIONS.topRight, "translate(-100%, ".concat(anchorSize, "px)")), _defineProperty(_FLOATING_MARKER_ANCH, _constants.FLOATING_MARKER_ANCHOR_POSITIONS.bottomLeft, "translate(".concat(anchorSize, "px, -100%)")), _defineProperty(_FLOATING_MARKER_ANCH, _constants.FLOATING_MARKER_ANCHOR_POSITIONS.bottomRight, "translate(-100%, -100%)"), _FLOATING_MARKER_ANCH)[anchor];
};

exports.getAnchorTransform = getAnchorTransform;
var DragShadowContainer = (0, _index.styled)('div', function (_ref) {
  var $theme = _ref.$theme,
      $height = _ref.$height,
      $width = _ref.$width,
      $dragging = _ref.$dragging;
  return {
    width: "".concat($width, "px"),
    height: "".concat($height, "px"),
    opacity: $dragging ? 1 : 0,
    visibility: $dragging ? 'visible' : 'hidden',
    transition: "".concat($theme.animation.timing300, " ").concat($theme.animation.easeOutCurve, " all"),
    position: 'relative',
    boxShadow: $theme.lighting.shadow600
  };
});
exports.DragShadowContainer = DragShadowContainer;
DragShadowContainer.displayName = "DragShadowContainer";
var DragShadow = (0, _index.styled)('div', function (_ref2) {
  var $theme = _ref2.$theme,
      $background = _ref2.$background,
      $width = _ref2.$width;
  return {
    background: $background,
    borderRadius: '50%',
    width: "".concat($width, "px"),
    height: "".concat(4, "px"),
    position: 'absolute',
    bottom: 0
  };
});
exports.DragShadow = DragShadow;
DragShadow.displayName = "DragShadow";
var Needle = (0, _index.styled)('div', function (_ref3) {
  var $theme = _ref3.$theme,
      $background = _ref3.$background,
      $height = _ref3.$height;
  return {
    background: $background,
    width: '4px',
    height: "".concat($height, "px"),
    boxShadow: $theme.lighting.shadow600
  };
});
exports.Needle = Needle;
Needle.displayName = "Needle";
var FloatingMarkerRoot = (0, _index.styled)('div', function () {
  return {
    position: 'relative',
    height: "".concat(_constants.xSmallPinheadDimension.height, "px"),
    width: "".concat(_constants.xSmallPinheadDimension.height, "px")
  };
});
exports.FloatingMarkerRoot = FloatingMarkerRoot;
FloatingMarkerRoot.displayName = "FloatingMarkerRoot";
var FloatingMarkerPinHeadContainer = (0, _index.styled)('div', function (_ref4) {
  var $theme = _ref4.$theme,
      $anchor = _ref4.$anchor,
      $anchorSize = _ref4.$anchorSize;
  return {
    position: 'absolute',
    transition: "".concat($theme.animation.timing300, " ").concat($theme.animation.easeOutCurve, " all"),
    transform: getAnchorTransform($anchor, $anchorSize)
  };
});
exports.FloatingMarkerPinHeadContainer = FloatingMarkerPinHeadContainer;
FloatingMarkerPinHeadContainer.displayName = "FloatingMarkerPinHeadContainer";
var FloatingMarkerAnchorContainer = (0, _index.styled)('div', function () {
  return {
    position: 'absolute'
  };
});
exports.FloatingMarkerAnchorContainer = FloatingMarkerAnchorContainer;
FloatingMarkerAnchorContainer.displayName = "FloatingMarkerAnchorContainer";
var FixedMarkerRoot = (0, _index.styled)('div', function () {
  return {
    display: 'flex',
    alignItems: 'center',
    flexDirection: 'column'
  };
});
exports.FixedMarkerRoot = FixedMarkerRoot;
FixedMarkerRoot.displayName = "FixedMarkerRoot";
var FixedMarkerDragContainer = (0, _index.styled)('div', function (_ref5) {
  var $theme = _ref5.$theme,
      $translateAmount = _ref5.$translateAmount,
      $performTranslate = _ref5.$performTranslate;
  return {
    transform: "translateY(".concat($performTranslate ? '0px' : "".concat($translateAmount, "px"), ")"),
    transition: "".concat($theme.animation.timing300, " ").concat($theme.animation.easeOutCurve, " all"),
    display: 'flex',
    alignItems: 'center',
    flexDirection: 'column'
  };
});
exports.FixedMarkerDragContainer = FixedMarkerDragContainer;
FixedMarkerDragContainer.displayName = "FixedMarkerDragContainer";
var OuterXSmallAnchor = (0, _index.styled)('div', function (_ref6) {
  var $theme = _ref6.$theme,
      $round = _ref6.$round,
      $background = _ref6.$background,
      $size = _ref6.$size;
  return {
    background: $background,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    height: "".concat($size, "px"),
    width: "".concat($size, "px"),
    borderRadius: $round ? '50%' : 0,
    boxShadow: $theme.lighting.shadow600
  };
});
exports.OuterXSmallAnchor = OuterXSmallAnchor;
OuterXSmallAnchor.displayName = "OuterXSmallAnchor";
var InnerXSmallAnchor = (0, _index.styled)('div', function (_ref7) {
  var $round = _ref7.$round,
      $color = _ref7.$color,
      $size = _ref7.$size;
  return {
    background: $color,
    height: "".concat($size, "px"),
    width: "".concat($size, "px"),
    borderRadius: $round ? '50%' : 0
  };
});
exports.InnerXSmallAnchor = InnerXSmallAnchor;
InnerXSmallAnchor.displayName = "InnerXSmallAnchor";
var PinHead = (0, _index.styled)('div', function (_ref8) {
  var $theme = _ref8.$theme,
      $height = _ref8.$height,
      $background = _ref8.$background,
      $gridTemplateColumns = _ref8.$gridTemplateColumns,
      $type = _ref8.$type,
      $forceCircle = _ref8.$forceCircle;
  var sharedStyles = {
    fixed: {
      padding: '0px 12px',
      borderRadius: "".concat($height, "px")
    },
    floating: {
      padding: '0px 8px'
    }
  };
  return _objectSpread(_objectSpread({
    background: $background,
    height: "".concat($height, "px"),
    display: 'grid',
    gridTemplateColumns: $gridTemplateColumns,
    gap: '8px',
    boxShadow: $theme.lighting.shadow600,
    whiteSpace: 'nowrap'
  }, sharedStyles[$type]), $forceCircle && {
    width: "".concat($height, "px"),
    display: 'flex',
    justifyContent: 'center',
    boxSizing: 'border-box'
  });
});
exports.PinHead = PinHead;
PinHead.displayName = "PinHead";
